packages/vscode-apollo/README.md

```diff
@@ -1,18 +1,154 @@
+- Allows [switching schema tags](#commands) to work on upcoming features
+- And more...
 
-## Configuration
+<img src="./images/marketplace/jump-to-def.gif" width="80%" style="margin: 5%" alt="Using jump to definition on a fragment">
 
-Please refer to the [Editor Plugin](https://www.apollographql.com/docs/platform/editor-plugins.html) docs for configuration instructions.
+<h2 id="getting-started">Getting Started</h2>
 
-## Issues
+To get started, first **install the Apollo GraphQL extension**. After installation, GraphQL syntax highlighting should automatically be enabled for `.graphql`, `.gql`, `.js` and `.ts` file types.
 
-Please report any issues you find to the [Apollo Tooling](https://github.com/apollographql/apollo-tooling) repo. Pull requests are always welcome!
+<h3 id="linking-a-schema">Linking a schema</h2>
+
+To get all of the benefits of the VS Code experience, its best to link the schema that is being developed against. The best way to do that is by [publishing a schema](./schema-registry.html#publish) to the Apollo schema registry. Once that is done, two steps are needed:
+
+1. Create an `apollo.config.js` at the root of the project
+2. Copy an API key from the Engine dashboard of the published service
+
+<h3 id="apollo-config">Setting up an Apollo config</h3>
+In order for the VS Code plugin to know how to find the schema, it needs to be linked to either a published schema or a local one. To link a project to a published schema, edit the `apollo.config.js` file to look like this:
+
+```js
+module.exports = {
+  client: {
+    service: "my-graphql-app"
+  }
+};
+```
+
+The service name is the id of the service created in Engine and can be found in the services dashboard of [Engine](https://engine.apollographql.com)
+
+> Important note: If the name of the service in Engine is changed, this value should be the service id. This can be found in the url when browsing the service in Engine. This will be easier to manage in the near future
+
+<h3 id="api-key">Setting up an API key</h3>
+To authenticate with Engine to pull down the schema, create a file next to the `apollo.config.js` called `.env`. This should be an untraced file (i.e. don't push it to GitHub). Go to the settings page of the published service and create a new API key.
+
+> It is best practice to create a new API key for each member of the team and name the key so its easy to find and revoke if needed
+
+After the key is found, add the following line to the `.env` file:
+
+```bash
+ENGINE_API_KEY=<enter copied key here>
+```
+
+After this is done, VS Code can be restarted and the editor integration will start providing autocomplete, validation, and more!
+
+<h3 id="local-schemas">Local schemas</h3>
+
+Sometimes it may make sense to link the editor to a locally running version of a schema to try out new designs that are in active development. To do this, the `apollo.config.js` file can be linked to a local service definition:
+
+```js
+module.exports = {
```

Comment:
This is a different approach than what you mention above.

A locally running server is equivalent to a "remote endpoint"
A local service definition file is the `localSchemaFile` key.